src/pages/settings/AboutPage/ShareLogList/BaseShareLogList.tsx

```diff
@@ -1,66 +1,62 @@
-import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
-import {View} from 'react-native';
+import React, {useMemo, useRef} from 'react';
 import {withOnyx} from 'react-native-onyx';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
-import {usePersonalDetails} from '@components/OnyxProvider';
-import OptionsSelector from '@components/OptionsSelector';
+import {useBetas, usePersonalDetails} from '@components/OnyxProvider';
 import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useDebouncedState from '@hooks/useDebouncedState';
 import useLocalize from '@hooks/useLocalize';
 import useNetwork from '@hooks/useNetwork';
-import useThemeStyles from '@hooks/useThemeStyles';
 import * as FileUtils from '@libs/fileDownload/FileUtils';
 import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import type {Report} from '@src/types/onyx';
 import type {BaseShareLogListOnyxProps, BaseShareLogListProps} from './types';
 
-function BaseShareLogList({betas, reports, onAttachLogToReport}: BaseShareLogListProps) {
-    const [searchValue, setSearchValue] = useState('');
-    const [searchOptions, setSearchOptions] = useState<Pick<OptionsListUtils.GetOptions, 'recentReports' | 'personalDetails' | 'userToInvite'>>({
-        recentReports: [],
-        personalDetails: [],
-        userToInvite: null,
-    });
-
+function BaseShareLogList({reports, onAttachLogToReport}: BaseShareLogListProps) {
+    const [searchValue, debouncedSearchValue, setSearchValue] = useDebouncedState('');
     const {isOffline} = useNetwork();
     const {translate} = useLocalize();
-    const styles = useThemeStyles();
     const isMounted = useRef(false);
     const personalDetails = usePersonalDetails();
+    const betas = useBetas();
+
+    const searchOptions = useMemo(() => {
+        const isOptionsDataReady = ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(personalDetails);
```

Comment:
Should this be outside of the function and then used as a dependency for `useMemo()`? It seems wrong to be calling it inside this memo since it relies on outside data.

Reply:
Also, why is `useMemo()` being done here instead of `useEffect()`?

Reply:
1. That would great but `OptionsListUtils.isPersonalDetailsReady(personalDetails);` is consuming data coming from the `usePersonalDetailsHook` so it can't be moved outside the component. 
2. I just tossed it in the useMemo responsible for `searchOptions` as it's also dependant on `personalDetails`. Using `useEffect` with `setState` is an anti-pattern that should punished by death in most cases.

Reply:
> so it can't be moved outside the component.

Sorry, this isn't really what I was suggesting. I was thinking of just moving it outside of `useMemo()` but still keeping it inside the component. I'd still like to see if this is possible.

> Using useEffect with setState is an anti-pattern that should punished by death in most cases.

Sounds good! I'll avoid the death penalty in that case then.